sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -415,6 +415,10 @@ def _parse_window_spec(spec):
           'Flatten': Flatten,
           'WindowInto': WindowInto,
           'GroupByKey': beam.GroupByKey,
+          'SumPerKey': lambda: beam.CombinePerKey(sum),
```

Comment:
Oh, I see you're adding them here. 

Would it be better to add a CombinePerKey (that takes a combine_fn "callable")? I'm a bit wary of adding something like TopNLargest just for this example. But I suppose right now for aggregation all we offer is SQL.

Reply:
Given everything is put into the global namespace, I think we should be judicious. Note that TopNLargest is just CombineGlobally(TopN).

Reply:
What would you suggest for this PR? Implement a more generic transform, such as CombineGlobally, instead of defining `TopNLargest`? What arguments would the CombineGlobally take in this case?

Reply:
CombineGlobally would take a TopN combiner.

Reply:
Two things:
1) I wanted to implement the CombineGlobally transform that would take a TopN combiner, as you suggested. For some reason I can't get it to work... That's my code for defining the transform in the dictionary in the yaml_provider.py:
        
```python
'CombineGlobally': lambda fn: beam.CombineGlobally(
              fn=python_callable.PythonCallableWithSource(fn)),
``` 
Then in the yaml pipeline I would use it like this:

```
  - type: CombineGlobally
  name: MostPopular
  input: XXX
  fn: "beam.combiners.TopCombineFn(1)"
```
  
but that results in error: 
  
```
self = <apache_beam.utils.python_callable.PythonCallableWithSource object at 0x7f9802d93b20>
args = ([3, 4],), kwargs = {}

    def __call__(self, *args, **kwargs):
>     return self._callable(*args, **kwargs)
E     TypeError: 'TopCombineFn' object is not callable
```

Do you have a suggestion how to implement it otherwise?



2) I still think that we should have high level transforms that are popular, like 'TopN', 'Largest', 'Smallest', etc available in the global scope. If the yaml api is aimed at less experienced developer, they will have a hard time creating these transforms with callable arguments.

Reply:
Hi @amotley and @robertwb, could you take a look at this unresolved issue?

Reply:
The new version of the notebook doesn't use this transform. Closing this comment.